### ISSUE TITLE
Custom icon file extension

### ISSUE THREAD

**Issue by plyoung** [rsubtil/controller_icons#87]
It would be nice if you could add an option to settings to specify a custom file extension. This would allow for a texture atlas to be used for example since I can then pass `.tres` as the extension, rather than using the default `.png`.

```
## Custom generic joystick mapper script
@export var custom_mapper : Script

## Icons image file extension
@export var file_extension : String = "png"
```

I did a quick hack by adding the setting and updating two spots in ControllerIcons.gd where it points to .png to make use of the setting and it seems to work at runtime so far. 

SpecificPathSelector.gd
```
func handle_files(category: String, base_path: String):
	for file in DirAccess.get_files_at(base_path):
		if file.get_extension() == ControllerIcons._settings.file_extension: #if file.get_extension() == "png":
			create_icon(category, base_path.path_join(file))
```

ControllerIcons.gd
```
204: base_path += path + "." + _settings.file_extension #base_path += path + ".png"
246: paths.push_back(base_path + "." + _settings.file_extension) #paths.push_back(base_path + ".png")
```

![Screenshot 2024-05-04 140448](https://github.com/rsubtil/controller_icons/assets/837362/0bb7d78e-d825-40f8-b07b-e020f034aa53)

**Comment by rsubtil** [comment]
I think this is a useful feature indeed! I've also been considering the future need of having to support other popular formats such as SVGs, so this would already handle that scenario as well.

I agree with the way you've patched the code to support this, honestly that's exactly how I'd do it. Could you go ahead and open a pull request with this feature?

**Comment by rsubtil** [comment]
Implemented with #88
